pkg/workload/cli/run.go

```diff
@@ -140,6 +140,7 @@ func init() {
 			var genFlags *pflag.FlagSet
 			if f, ok := gen.(workload.Flagser); ok {
 				genFlags = f.Flags().FlagSet
+				gen.(workload.Flagser).Flags().AddFlag(runFlags.Lookup("duration"))
```

Comment:
I'm not sure if this is doing anything. We add genFlags down below alongside all of runFlags to the run command already?

Reply:
Flags added to runFlags are not propagated to the workload flags are not accessible in the workload. That's why we need to add the `duration` flag to the generator flags.